packages/cdk8s-plus-17/src/container.ts

```diff
@@ -285,6 +285,10 @@ export class Container {
   private readonly _startup?: Probe;
 
   constructor(props: ContainerProps) {
+    if (props instanceof Container) {
```

Comment:
How can this happen if the type of `props` is a ContainerProps?

Reply:
The test suite below shows an example.  The issue is that a Container object is considered a valid ContainerProps interface.  The reason is that anything is a valid ContainerProps that has an `image: string` property.  (see: https://github.com/awslabs/cdk8s/blob/aa466d46757e1e856c7c61710abef8424f53b9ef/packages/cdk8s-plus-17/src/container.ts#L149)